apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ClientPool.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Api.Gax.Grpc;
+using Google.Apis.Auth.OAuth2;
+using Grpc.Auth;
+
+namespace Google.Cloud.Spanner.V1
+{
+    internal static class ClientPool
+    {
+        static readonly ConcurrentDictionary<ITokenAccess, ClientEntry> ClientEntryPool =
+                new ConcurrentDictionary<ITokenAccess, ClientEntry>();
+
+        public static async Task<SpannerClient> AcquireClientAsync(ITokenAccess credentials)
```

Comment:
We probably need to think about what kind of equality we get from `ITokenAccess` in various cases.